### DISCUSSION TITLE
Working with IFG

### DISCUSSION THREAD

**Discussion by locdaika** [rdkit/rdkit#6607] [Q&A] [answered]
Dear @ricrogz,

thank you for your code on IFG, expressing the paper by Peter Ertl.

I tried running the code on my own, yet, it does not work for me, it does not print in the end.

Could you please help with?
Best,
Dac-Loc

**Comment by ricrogz** [comment]
Hi @locdaika, I'm afraid I'm not the one that wrote that code. I'm not familiar with IFG at all.

**Reply by locdaika** [reply]
@ricrogz : thank you for your quick response. :)

**Comment by thegodone** [comment, marked as answer]
Hi @locdaika,

I wrote it.

Can you give me an example code you use ?

It is hard to debug without a context.

thanks,

Br

Guillaume

**Reply by locdaika** [reply]
Greetings @thegodone and @ricrogz, 

thank you a lot for your quick response. 

@thegodone : I am sorry. My bad, I have tried running again and printed results into a text file already. 

Another question:
Could you please suggest marking and printing the 20 given sample molecules? or is it impossible to execute with Python but only with Java?

**Comment by thegodone** [comment]
Hi,

Just copy Contrib/ifg.py file in one of your folders, be sure rdkit is
installed and activated (conda or pip install) and conda activate...

than open the terminal and run the following command (where you have copied
ifg.py file):
# python ifg.py
the output should be:
 1: 4 fgs [IFG(atomIds=(2,), atoms='n', type='cnc'), IFG(atomIds=(4, 5, 6,
7), atoms='NS(=O)=O', type='cNS(c)(=O)=O'), IFG(atomIds=(12,), atoms='N',
type='cN'), IFG(atomIds=(15,), atoms='n', type='cnc')]
 2: 4 fgs [IFG(atomIds=(0, 1, 2), atoms='N=CN', type='cC(=N)N'),
IFG(atomIds=(8, 7), atoms='C=C', type='cC=Cc'), IFG(atomIds=(15,),
atoms='O', type='cO'), IFG(atomIds=(16, 17, 18), atoms='N=CN',
type='cC(=N)N')]
 3: 4 fgs [IFG(atomIds=(1, 2, 3), atoms='NC=O', type='cNC(C)=O'),
IFG(atomIds=(5, 6), atoms='nn', type='cnnc'), IFG(atomIds=(8,), atoms='s',
type='csc'), IFG(atomIds=(9, 10, 11, 12), atoms='NS(=O)=O',
type='cS(N)(=O)=O')]
 4: 3 fgs [IFG(atomIds=(0, 1, 2, 3), atoms='NS(=O)=O', type='cS(N)(=O)=O'),
IFG(atomIds=(8, 9, 10, 11, 12, 13), atoms='NCNS(=O)=O',
type='O=S1(=O)ccNCN1'), IFG(atomIds=(16,), atoms='Cl', type='cCl')]
 5: 3 fgs [IFG(atomIds=(1, 2, 3), atoms='N=CN', type='cN=C(C)NC'),
IFG(atomIds=(8,), atoms='Cl', type='cCl'), IFG(atomIds=(11, 12, 13),
atoms='C=[N+][O-]', type='cC(c)=[N+](C)[O-]')]
 6: 4 fgs [IFG(atomIds=(2, 3), atoms='no', type='cnoc'), IFG(atomIds=(8, 6,
7), atoms='NC=O', type='cC(=O)NC'), IFG(atomIds=(16, 17, 18, 10, 11),
atoms='O=CNCS', type='O=C1CC2SCCN12'), IFG(atomIds=(19, 20, 21),
atoms='O=CO', type='CC(=O)O')]
 7: 4 fgs [IFG(atomIds=(0,), atoms='Cl', type='cCl'), IFG(atomIds=(8, 7),
atoms='C=N', type='cC(c)=NC'), IFG(atomIds=(10, 11, 12), atoms='NC=O',
type='cNC(C)=O'), IFG(atomIds=(19, 20, 21), atoms='O=[N+][O-]',
type='c[N+](=O)[O-]')]
 8: 6 fgs [IFG(atomIds=(1,), atoms='O', type='cOC'), IFG(atomIds=(8, 9, 7),
atoms='NC=O', type='cC(=O)NC'), IFG(atomIds=(15,), atoms='N',
type='CN(C)C'), IFG(atomIds=(17, 18), atoms='C=C', type='C=CC'),
IFG(atomIds=(20,), atoms='O', type='cOC'), IFG(atomIds=(24, 25, 22, 23),
atoms='NS(=O)=O', type='cS(N)(=O)=O')]
 9: 4 fgs [IFG(atomIds=(5,), atoms='Cl', type='cCl'), IFG(atomIds=(7,),
atoms='N', type='cNc'), IFG(atomIds=(16, 14, 15), atoms='O=CO',
type='cC(=O)O'), IFG(atomIds=(18,), atoms='Cl', type='cCl')]
10: 4 fgs [IFG(atomIds=(0,), atoms='Cl', type='cCl'), IFG(atomIds=(5,),
atoms='O', type='cOc'), IFG(atomIds=(12, 13, 14), atoms='N=CN',
type='cN=C(c)N(C)C'), IFG(atomIds=(17,), atoms='N', type='CNC')]
11: 6 fgs [IFG(atomIds=(0,), atoms='F', type='CF'), IFG(atomIds=(2,),
atoms='F', type='CF'), IFG(atomIds=(3,), atoms='F', type='CF'),
IFG(atomIds=(5, 6, 7), atoms='NC=O', type='cN(C)C(C)=O'), IFG(atomIds=(9,
10), atoms='C=N', type='cC(c)=NC'), IFG(atomIds=(20,), atoms='Cl',
type='cCl')]
12: 7 fgs [IFG(atomIds=(0,), atoms='O', type='CO'), IFG(atomIds=(3,),
atoms='O', type='COC'), IFG(atomIds=(7,), atoms='O', type='CO'),
IFG(atomIds=(8,), atoms='n', type='cn(c)C'), IFG(atomIds=(10,), atoms='n',
type='cnc'), IFG(atomIds=(13,), atoms='O', type='CO'), IFG(atomIds=(16, 17,
15), atoms='N=CN', type='cN=CNC')]
13: 4 fgs [IFG(atomIds=(2,), atoms='N', type='CNC'), IFG(atomIds=(8, 9, 7),
atoms='O=S=O', type='cS(C)(=O)=O'), IFG(atomIds=(11,), atoms='s',
type='csc'), IFG(atomIds=(16, 17, 18, 15), atoms='NS(=O)=O',
type='cS(N)(=O)=O')]
14: 4 fgs [IFG(atomIds=(2,), atoms='O', type='CO'), IFG(atomIds=(7, 8, 9,
10, 11, 12, 13, 14, 15), atoms='O=CNC(=CS)C(=O)O',
type='CSC1=C(C(=O)O)N2C(=O)CC2C1'), IFG(atomIds=(18,), atoms='N',
type='CNC'), IFG(atomIds=(21, 22, 23), atoms='NC=O', type='CC(=O)N(C)C')]
15: 8 fgs [IFG(atomIds=(3,), atoms='N', type='cN(C)C'), IFG(atomIds=(7,),
atoms='N', type='CNC'), IFG(atomIds=(10,), atoms='F', type='cF'),
IFG(atomIds=(12,), atoms='N', type='cN'), IFG(atomIds=(15,), atoms='O',
type='c=O'), IFG(atomIds=(18,), atoms='n', type='cn(c)C'),
IFG(atomIds=(24,), atoms='F', type='cF'), IFG(atomIds=(25, 26, 27),
atoms='O=CO', type='cC(=O)O')]
16: 2 fgs [IFG(atomIds=(1, 2), atoms='C=C', type='CC=C(C)C'),
IFG(atomIds=(5, 6, 7, 8, 9, 10), atoms='O=CNNC=O',
type='cN1C(=O)CC(=O)N1c')]
17: 3 fgs [IFG(atomIds=(0,), atoms='Cl', type='cCl'), IFG(atomIds=(5, 6, 7,
8, 9, 11), atoms='N=C(N)NC=O', type='cN=C1NC(=O)CN1C'), IFG(atomIds=(15,),
atoms='Cl', type='cCl')]
18: 6 fgs [IFG(atomIds=(1, 2, 3), atoms='NC=O', type='CNC(C)=O'),
IFG(atomIds=(8, 9, 6, 7), atoms='N=C(N)N', type='CNC(=N)N'),
IFG(atomIds=(20, 21, 22, 10, 11, 12), atoms='C=C(O)C(=O)O',
type='CC=C(OC)C(=O)O'), IFG(atomIds=(15,), atoms='O', type='CO'),
IFG(atomIds=(17,), atoms='O', type='CO'), IFG(atomIds=(19,), atoms='O',
type='CO')]
19: 8 fgs [IFG(atomIds=(2,), atoms='O', type='CO'), IFG(atomIds=(4,),
atoms='O', type='CO'), IFG(atomIds=(7,), atoms='N', type='cNC'),
IFG(atomIds=(9,), atoms='n', type='cnc'), IFG(atomIds=(11,), atoms='N',
type='cN'), IFG(atomIds=(12,), atoms='n', type='cnc'), IFG(atomIds=(14,),
atoms='O', type='cO'), IFG(atomIds=(16,), atoms='N', type='cNC')]
20: 7 fgs [IFG(atomIds=(0,), atoms='N', type='CN'), IFG(atomIds=(6,),
atoms='N', type='cN(C)C'), IFG(atomIds=(10,), atoms='Cl', type='cCl'),
IFG(atomIds=(14,), atoms='O', type='c=O'), IFG(atomIds=(17,), atoms='n',
type='cn(c)C'), IFG(atomIds=(23,), atoms='Cl', type='cCl'),
IFG(atomIds=(24, 25, 26), atoms='O=CO', type='cC(=O)O')]
